org.jacoco.core/src/org/jacoco/core/internal/instr/InstrSupport.java

```diff
@@ -158,18 +158,16 @@ private InstrSupport() {
 	 */
 	static final int CLINIT_ACC = Opcodes.ACC_SYNTHETIC | Opcodes.ACC_STATIC;
 
-	private static final int MAJOR_VERSION_INDEX = 6;
-
 	/**
-	 * Gets major of bytecode version number from given bytes of class.
+	 * Gets major of bytecode version number from given {@link ClassReader}.
 	 *
-	 * @param b
-	 *            bytes of class
-	 * @return version of bytecode
+	 * @param reader
+	 *            reader to get information about the class
+	 * @return major version of bytecode
 	 */
-	public static int getVersionMajor(final byte[] b) {
-		return (short) (((b[MAJOR_VERSION_INDEX] & 0xFF) << 8)
-				| (b[MAJOR_VERSION_INDEX + 1] & 0xFF));
+	public static int getVersionMajor(final ClassReader reader) {
+		final int firstConstantPoolEntryOffset = reader.getItem(1) - 1;
```

Comment:
Why are you going relative to the contant pool entry? Isn't it always at the same absolute position? I'm asking because we have another method `classReaderFor()` in the same class which uses a fixed lookup index. Should be consistent.

Reply:
@marchof recall #850 - there is constructor [`ClassReader(byte[] classFileBuffer, int classFileOffset, int classFileLength)`](https://gitlab.ow2.org/asm/asm/blob/ASM_7_1/asm/src/main/java/org/objectweb/asm/ClassReader.java#L165-177).

Reply:
@marchof so relative to the constant pool entry seems safer to me, even if currently our code does not pass such instances of `ClassReader` into this method.

Reply:
I can update unit test to show/cover this case.

Or we can create subclass of `ClassReader` that will returned by `classReaderFor` and required here.

Reply:
@Godin ASM API is really wired here: it has public `readX()` methods but keeps the offset private. My proposals to resolve this:

* Use absolute positions as the offset constructor cannot be used any more due to #850 or
* Add a inline comment to explain why it is done this way